### PR TITLE
docs: Add additional information regarding performance fine-tuning

### PR DESCRIPTION
**Pull request by Kawron**
Updates do the sc4s documentation regarding fine-tuning performance.

### CODE REVIEW

File: docs/architecture/lb/f5.md [hunk 1]
```diff
@@ -139,6 +139,8 @@ Follow the [Check TCP Performance](../performance-tests.md#check-your-tcp-perfor
 - **SC4S Instances**: `c5.4xlarge`  
 - **F5 Instance**: `r5.8xlarge, 32vCPU, 256GiB Memory, 10Gbps Network Performance`  
 
+<small>Please note that the performance test results are based on previous versions of SC4S. We will update the results to reflect the latest SC4S version performance in the future.</small>
```
Comment (sbylica-splunk) [inline docs/architecture/lb/f5.md]: Which vesrions? I would say something more generic, like "Performance may vary depending on a version and specifics of your environment". Same for all the ones below

File: docs/architecture/fine-tuning.md [hunk 2]
```diff
@@ -15,32 +17,94 @@ This section provides guidance on improving SC4S performance by tuning configura
...
+In the example above, the input window can store up to **100,000 messages**. Note that increasing the window size will increase the application's memory usage.
+
+### Fetch limit
+
+When increasing the input window size, you may also need to increase the **fetch limit**. The fetch limit controls the maximum number of messages retrieved from the source in a single read operation.
+
+- **Too high**: Should not exceed the input window size, as this would fill the entire buffer in one read cycle.
+- **Too low**: Results in underutilizing the buffer capacity, requiring more read cycles to process the same volume.
+
+The default value is `1000`.
```
Comment (sbylica-splunk) [inline docs/architecture/fine-tuning.md:105]: Maybe we could mention diskbuffer here?

File: docs/architecture/lb/f5.md (hunk 1 above)
Comment (Kawron) [reply docs/architecture/lb/f5.md]: You are right, I think the more generic version is better I will change it

File: docs/architecture/fine-tuning.md (hunk 2 above)
Comment (Kawron) [reply docs/architecture/fine-tuning.md:105]: Good idea, done

File: docs/architecture/lb/f5.md [hunk 3]
```diff
@@ -139,6 +139,8 @@ Follow the [Check TCP Performance](../performance-tests.md#check-your-tcp-perfor
 - **SC4S Instances**: `c5.4xlarge`  
 - **F5 Instance**: `r5.8xlarge, 32vCPU, 256GiB Memory, 10Gbps Network Performance`  
 
+<small>Performance may vary depending on a version and specifics of your environment.</small>
```
Comment (jplatosz-splunk) [inline docs/architecture/lb/f5.md]: avoid using html

File: docs/architecture/lb/f5.md
Comment (jplatosz-splunk) [inline docs/architecture/lb/f5.md]: notes should have a different formatting such as:
!!! note "Note"
   Lorem ipsum ....

File: docs/architecture/lb/f5.md [hunk 4]
```diff
@@ -200,6 +202,8 @@ Follow the [Check UDP Performance](../performance-tests.md#check-your-udp-perfor
...
+
 | Receiver / Drops Rate for EPS (msgs/sec) | 4,500  | 9,000  | 27,000 | 50,000 | 150,000 |
```
Comment (jplatosz-splunk) [inline docs/architecture/lb/f5.md]: Receiver / Drops rate for EPS

File: docs/architecture/fine-tuning.md [hunk 5]
```diff
@@ -0,0 +1,203 @@
+# Finetune SC4S
+
+This section provides guidance on improving SC4S performance by tuning configuration settings.
+
+You can apply these settings to your infrastructure to improve SC4S performance. After making adjustments, run the [performance tests](performance-tests.md#check-your-tcp-performance) and retain the changes that result in performance improvements.
+
+## Disabling some of the features
```
Comment (jplatosz-splunk) [inline docs/architecture/fine-tuning.md]: ## Disable features that reduce performance

File: docs/architecture/fine-tuning.md [hunk 6]
```diff
@@ -0,0 +1,203 @@
...
+Some SC4S features may negatively impact performance. If you experience performance issues, consider disabling some of the settings like **name cache** and **message grouping**:
+
+```bash
+SC4S_USE_NAME_CACHE=no
+SC4S_SOURCE_VMWARE_VSPHERE_GROUPMSG=no
+```
+
+## Dedicated sc4s instances
+
+If one of the logs sources produces a large percentage of the overall traffic, it is advised to create an additional dedicated sc4s service on a separate host.
```
Comment (jplatosz-splunk) [inline docs/architecture/fine-tuning.md]: If one of the logs sources produces a large percentage of the overall traffic, create an additional dedicated sc4s service on a separate host.

Note from J: we avoid in docs terms like "it is advised", "we recommend" etc

File: docs/architecture/fine-tuning.md [hunk 7]
```diff
@@ -0,0 +1,203 @@
...
+
+## Tune the receiving buffer
+
+Increasing the receive buffer allows the kernel to queue more incoming data before the application processes it, reducing packet loss during traffic bursts. This requires changes at both the OS level and within SC4S.
+
+### Step 1: Increase the kernel buffer OS limits
+
+Edit `/etc/sysctl.conf` and set the receive buffer size to 512 MB:
```
Comment (jplatosz-splunk) [inline docs/architecture/fine-tuning.md]: Make make it into steps, like


1. Edit ....
2. Apply the changes ....

File: docs/architecture/fine-tuning.md [hunk 8]
```diff
@@ -0,0 +1,203 @@
...
+
+## Dedicated sc4s instances
+
+If one of the logs sources produces a large percentage of the overall traffic, it is advised to create an additional dedicated sc4s service on a separate host.
+
+## Tune the receiving buffer
+
+Increasing the receive buffer allows the kernel to queue more incoming data before the application processes it, reducing packet loss during traffic bursts. This requires changes at both the OS level and within SC4S.
+
+### Step 1: Increase the kernel buffer OS limits
```
Comment (jplatosz-splunk) [inline docs/architecture/fine-tuning.md]: I think we can remove this Step 1 from the procedure name, and maybe instead move it to an introductory sentence below the title, sth like: 

Start turning the receiving buffer with increasing the kernel buffer OS limits. Perform the following steps to change the buffer size:

File: docs/architecture/fine-tuning.md [hunk 9]
```diff
@@ -0,0 +1,203 @@
...
+net.core.rmem_max = 536870912
+```
+
+Apply the changes:
+
+```bash
+sudo sysctl -p
+```
+
+### Step 2: Configure SC4S to use the larger buffer
```
Comment (jplatosz-splunk) [inline docs/architecture/fine-tuning.md]: You can change this step and the following in the similar way as above

File: docs/architecture/fine-tuning.md [hunk 10]
```diff
@@ -0,0 +1,203 @@
...
+
+```bash
+SC4S_SOURCE_TCP_SO_RCVBUFF=536870912
+```
+
+### Step 3: Restart SC4S
+
+Restart SC4S for the changes to take effect.
+
+### Other Protocols
```
Comment (jplatosz-splunk) [inline docs/architecture/fine-tuning.md]: Other protocols

File: docs/architecture/fine-tuning.md [hunk 11]
```diff
@@ -0,0 +1,203 @@
...
+
+Input window provides flow‑control at the application level. Syslog‑ng uses this feature to temporarily buffer messages when outputs are slow. The mechanism works by pulling messages from the kernel’s receive buffer and placing them into an application buffer.
+
+- The window size defines how many messages this internal buffer can hold.
+- Each message fetched from the kernel buffer increases the window counter.
+- Each message successfully forwarded to the output decreases the counter.
+
+To change the window size, modify the following options in `/opt/sc4s/env_file`:
+
+### For TCP
```
Comment (jplatosz-splunk) [inline docs/architecture/fine-tuning.md]: since its a part of the procedure, I'd just make it  bolded instead of a heading
**For TCP**

File: docs/architecture/fine-tuning.md [hunk 12]
```diff
@@ -0,0 +1,203 @@
...
+
+```bash
+SC4S_SOURCE_TCP_IW_SIZE=1000000
+```
+
+### For UDP
```
Comment (jplatosz-splunk) [inline docs/architecture/fine-tuning.md]: and here the same - **For UDP**

File: docs/architecture/fine-tuning.md [hunk 13]
```diff
@@ -0,0 +1,203 @@
...
+### Fetch limit
+
+When increasing the input window size, you may also need to increase the **fetch limit**. The fetch limit controls the maximum number of messages retrieved from the source in a single read operation.
+
+- **Too high**: Should not exceed the input window size, as this would fill the entire buffer in one read cycle.
+- **Too low**: Results in underutilizing the buffer capacity, requiring more read cycles to process the same volume.
+
+The default value is `1000`.
+
+## Disk Buffering
```
Comment (jplatosz-splunk) [inline docs/architecture/fine-tuning.md]: Disk buffering

File: docs/architecture/fine-tuning.md [hunk 14]
```diff
@@ -0,0 +1,203 @@
...
+
+To prevent message loss during HEC connection outages, consider enabling [Disk Buffering](../configuration.md#configure-your-sc4s-disk-buffer). This feature temporarily stores messages on disk when the destination is unavailable.
+
+## Switch to SC4S Lite
+
+Parsing syslog messages can be a CPU-intensive task. During the parsing process, each syslog message goes through multiple parsing rules until a match is found. Some log messages follow longer parsing paths than others, and some parsers use regular expressions, which can be slow.
+
+If you are familiar with your log sources, consider performing an A/B test and switching to SC4S Lite, which includes only the parsers for the vendors you require. Although artificial performance tests may not fully reflect the impact of this change, you may observe an increase in the capacity of your syslog layer when operating with real-world data.
+
+## Finetune for UDP Traffic
```
Comment (jplatosz-splunk) [inline docs/architecture/fine-tuning.md]: Finetune for UDP traffic

File: docs/architecture/fine-tuning.md [hunk 15]
```diff
@@ -0,0 +1,203 @@
...
+
+### Tested configuration:
+- **Loggen** - c5.2xlarge
+- **SC4S** (3.29.0) + podman - c5.4xlarge
+- **Splunk Cloud** 9.2.2403.105 - 30IDX
+
+<small>Performance may vary depending on a version and specifics of your environment.</small>
```
Comment (jplatosz-splunk) [inline docs/architecture/fine-tuning.md]: remove this tag

File: docs/architecture/fine-tuning.md [hunk 16]
```diff
@@ -0,0 +1,203 @@
...
+
+| Setup for 67,000 EPS (Events per Second) | % Loss |
+|------------------------------------------|--------|
+| Default                                  | 77.88  |
+| OS Kernel Tuning                         | 24.38  |
+| Increasing the Number of UDP Sockets     | 22.95  |
```
Comment (jplatosz-splunk) [inline docs/architecture/fine-tuning.md:131]: Number of UDP Sockets  - this it have to be capitalized?

File: docs/architecture/fine-tuning.md [hunk 17]
```diff
@@ -0,0 +1,203 @@
...
+### Parallelize TCP processing
+1. Update `/opt/sc4s/env_file` and restart SC4S.
+```
+SC4S_ENABLE_PARALLELIZE=yes
+SC4S_PARALLELIZE_NO_PARTITION=4
+```
+
+Parallelize distributes messages from a single TCP stream across multiple concurrent threads, which is noticeable in production environments with a single high-volume TCP source.
+
+| SC4S Parallelize    | Loggen TCP Connections         | %Cpu(s) us | Average Rate (msg/sec) |
```
Comment (jplatosz-splunk) [inline docs/architecture/fine-tuning.md]: | SC4S parallelize    | Loggen TCP connections         | % CPUs used | Average rate (msg/sec) |

File: docs/architecture/performance-tests.md [hunk 18]
```diff
@@ -54,14 +54,15 @@ Here is a reference example of performance testing using our lab configuration o
...
 * SC4S(2.30.0) + podman (4.0.2) - m5zn family
-* SC4S_DEST_SPLUNK_HEC_DEFAULT_WORKERS=10 (default)
 * Splunk Cloud Noah 8.2.2203.2 - 3SH + 3IDX
 
 ### Command
 ```bash
 /opt/syslog-ng/bin/loggen -i --rate=100000 --interval=1800 -P -F --sdata="[test name=\"stress17\"]" -s 800 --active-connections=10 <local_hostmane> <sc4s_external_tcp514_port>
 ```
 
+<small>Performance may vary depending on a version and specifics of your environment.</small>
```
Comment (jplatosz-splunk) [inline docs/architecture/performance-tests.md]: same as above

File: docs/configuration.md [hunk 19]
```diff
@@ -130,31 +130,31 @@ therefore an administrator must provide a means of log rotation to prune files a
...
-| SC4S_SOURCE_LISTEN_RFC5425_SOCKETS | 1                  | Number of kernel sockets per active UDP port, which configures multi-threading of the input buffer in the kernel to prevent packet loss. Total UDP input buffer is the sum of SOCKETS x SO_RCVBUFF. |
+| Variable | Values/Default     | Description                                                                                                                                                                                                                                 |
+|----------|--------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| SC4S_SOURCE_TLS_ENABLE | yes or no(default) | Enable TLS globally.  Be sure to configure the certificate as shown below.                                                                                                                                                                  |
```
Comment (jplatosz-splunk) [inline docs/configuration.md]: We avoid words like above or below, instead you can use -  Be sure to configure the certificate as shown in the following section.

File: docs/configuration.md [hunk 20]
```diff
@@ -319,10 +319,10 @@ syslog-ng config code.
...
-|----------|---------------|-------------|
-| SC4S_ENABLE_EBPF=yes  | yes or no(default) | Use eBPF to leverage multithreading when consuming from a single connection. |
-|SC4S_EBPF_NO_SOCKETS=4 | integer | Set number of threads to use. For optimal performance this should not be less than the value set for  `SC4S_SOURCE_LISTEN_UDP_SOCKETS`. |
+| Variable | Values        | Description                                                                                                                                                               |
+|----------|---------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| SC4S_ENABLE_EBPF=yes  | yes or no(default) | Use eBPF to leverage multithreading when consuming from a single connection.                                                                                              |
+|SC4S_EBPF_NO_SOCKETS=4 | integer | Set number of threads to use. For optimal performance this should be set to 4 x number of cores but not less than the value set for  `SC4S_SOURCE_LISTEN_UDP_SOCKETS`. |
```
Comment (jplatosz-splunk) [inline docs/configuration.md]: For optimal performance, set it to 4 x number of cores but not less than the value set for  `SC4S_SOURCE_LISTEN_UDP_SOCKETS`.

File: mkdocs.yml [hunk 21]
```diff
@@ -41,9 +41,9 @@ nav:
       - Read First: "architecture/index.md"
       - Scaling Solutions:
           - Performance Tests: "architecture/performance-tests.md"
-          - Fine-Tuning:
-              - TCP Optimization: "architecture/tcp-optimization.md"
-              - UDP Optimization: "architecture/udp-optimization.md"
+          - Fine-Tuning: "architecture/fine-tuning.md"
+              # - TCP Optimization: "architecture/tcp-optimization.md"
```
Comment (jplatosz-splunk) [inline mkdocs.yml]: I was unable to preview the doc locally due to time-out error, I wonder why is # added to the toc ?

Comment (Kawron) [reply mkdocs.yml]: Oh, that's just a commented line. It should have been deleted, i will fix that

File: docs/architecture/fine-tuning.md (hunk 16 above)
Comment (Kawron) [reply docs/architecture/fine-tuning.md:131]: I guess it doesn't have to. In that case OS Kernel Tuning should be -> OS kernel tuning?

File: docs/architecture/fine-tuning.md (hunk 8 above)
Comment (Kawron) [reply docs/architecture/fine-tuning.md]: I think maybe we should get rid of ### for each step and instead do sth like this:

```
## Tune the receiving buffer

Increasing the receive buffer allows the kernel to queue more incoming data before the application processes it, reducing packet loss during traffic bursts. This requires changes at both the OS level and within SC4S. Start turning the receiving buffer with increasing the kernel buffer OS limits. Perform the following steps to change the buffer size:

1. Edit `/etc/sysctl.conf` and set the receive buffer size to 512 MB:

'''bash
net.core.rmem_default = 536870912
net.core.rmem_max = 536870912
'''

2. Apply the changes:

'''bash
sudo sysctl -p
'''

Next configure SC4S to use the larger buffer:

1. Add the following line to `/opt/sc4s/env_file`:

'''bash
SC4S_SOURCE_TCP_SO_RCVBUFF=536870912
'''

2. Restart SC4S for the changes to take effect.

3. Apply the same buffer tuning to each syslog transport you have enabled:

'''bash
SC4S_SOURCE_TCP_SO_RCVBUFF=536870912       # Generic syslog over TCP
SC4S_SOURCE_UDP_SO_RCVBUFF=536870912       # Generic syslog over UDP
SC4S_SOURCE_RFC5426_SO_RCVBUFF=536870912   # RFC 5426 (syslog over UDP)
SC4S_SOURCE_RFC6587_SO_RCVBUFF=536870912   # RFC 6587 (syslog over TCP)
SC4S_SOURCE_RFC5425_SO_RCVBUFF=536870912   # RFC 5425 (syslog over TLS)
'''
```

Then we won't have weird sections with only one instructions

File: docs/architecture/fine-tuning.md (hunk 7 above)
Comment (Kawron) [reply docs/architecture/fine-tuning.md]: see: https://github.com/splunk/splunk-connect-for-syslog/pull/2833#discussion_r2598665008

File: docs/architecture/fine-tuning.md (hunk 9 above)
Comment (Kawron) [reply docs/architecture/fine-tuning.md]: see: https://github.com/splunk/splunk-connect-for-syslog/pull/2833#discussion_r2598665008

File: docs/architecture/fine-tuning.md [hunk 22]
```diff
@@ -0,0 +1,199 @@
...
+
+This section provides guidance on improving SC4S performance by tuning configuration settings.
+
+You can apply these settings to your infrastructure to improve SC4S performance. After making adjustments, run the [performance tests](performance-tests.md#check-your-tcp-performance) and retain the changes that result in performance improvements.
+
+## Disable features that reduce performance
+
+Some SC4S features may negatively impact performance. If you experience performance issues, consider disabling some of the settings like **name cache** and **message grouping**:
+
+```bash
```
Comment (ajasnosz) [inline docs/architecture/fine-tuning.md:13]: It would be good to mention in which file those variables should be added, and if the sc4s should be restarted for it to take effect

Comment (Kawron) [reply docs/architecture/fine-tuning.md:13]: Done

**Review by sbylica-splunk — APPROVED** [review]
